src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/Event.scala

```diff
@@ -242,30 +243,56 @@ case class Event(
 
 object Event {
 
+  @deprecated("Event.unsafe functionality is merged into Event.parse method", "3.1.0")
   object unsafe {
-    implicit def unsafeEventDecoder: Decoder[Event] = deriveDecoder[Event]
+    implicit def unsafeEventDecoder: Decoder[Event] = Event.eventDecoder
   }
```

Comment:
I see, thanks for the explanation.

> An alternative option would be to release this as version 4.0.0 and then break binary compatibility. But I figured Snowplow users would probably prefer to minimize this kind of disruption.

Indeed. Although it will be weird to release 4.0.0 with this legacy code in it when the time comes (unless we think of removing it, but then why not releasing 4.0.0 now).